mbf_abstract_nav/src/controller_action.cpp

```diff
@@ -351,9 +353,21 @@ void ControllerAction::fillExePathResult(
   result.outcome = outcome;
   result.message = message;
   result.final_pose = robot_pose_;
-  result.dist_to_goal = static_cast<float>(mbf_utility::distance(robot_pose_, goal_pose_));
+  result.dist_to_goal = global_plan_distance_;
   result.angle_to_goal = static_cast<float>(mbf_utility::angle(robot_pose_, goal_pose_));
 }
 
+float ControllerAction::calculateGlobalPathLength(
+  GoalHandle& goal_handle)
+{
+  float plan_distance = 0;
+  const mbf_msgs::ExePathGoal &goal = *(goal_handle.getGoal().get());
+  const std::vector<geometry_msgs::PoseStamped> &plan = goal.path.poses;
+  for(int i = 0; i < plan.size() - 2; i++){
+    plan_distance += static_cast<float>(mbf_utility::distance(plan[i], plan[i+1]));
```

Comment:
looks like mbf_utility::distance returns a double. Why do we need float? Can we just change type to double even for dist_to_goal ? Also, plan_distance should be 0.0 if it is a float or double.

Reply:
The feedback msg wants it as a float